### ISSUE TITLE
Remove Google Analytics?

### ISSUE THREAD

**Issue by alphapapa** [200ok-ch/organice#41]
Hi,

Thanks for starting this project to make `org-web` more widely useful, and for putting your resources behind it.

Looking briefly at the code, the first thing I noticed was this:

https://github.com/200ok-ch/organice/blob/47f562e959b3ae5443dc55461ae872afb8b5c855/src/store.js#L3

As an Emacs/Org user, that's quite off-putting.  Were I to use and trust a project like this on my Org files, I'd want Google Analytics to be nowhere near any part of it.

Would you consider removing that from the code?

**Comment by munen** [comment]
Hi @alphapapa 

Happy to hear that you like the general direction of the project!

Regarding GA, I didn't know that it was integrated until a few days ago myself. I do understand that you don't like the integration at all. Personally, I'm no friend of Google as well!

Having said that, the first most important thing to mention is that no content is tracked (which otherwise would have been a *huuge* issue on org-web. The relevant code is here: https://github.com/200ok-ch/organice/blob/master/src/middleware/google_analytics_logging.js#L6

After I saw a couple of days ago that GA is integrated, I did change it to a new key, so that I can actually see the traffic. I did so, because organice still has lots of bugs and in theory it would be nice to see/track those in some form somewhere.

However, GA is certainly not the best place for that, so let me propose the following: I'll happily accept PRs with any of these features:

- Document the usage of GA in the README and make the actual integration GA configurable, so that anyone hosting her own version of organice can take it out. Making organice as easy to deploy as possible for anyone is one of the main goals.

- Replace GA with a FOSS alternative whilst having the same documentation and configuration option as above.

Would you be interested in doing one or the other?

**Comment by munen** [comment]
Let me rephrase the second option: I'm not actually interested in tracking people using organice, whatsoever! Therefore the second option doesn't even have to be product close to GA in functionality.

However, I'm interested in getting automated crash-reports to learn about all the things going wrong for the community so that we can start to get organice into a stable and productive state. Even sending the crash-reports should be an opt-in feature, imo.

Personally, I would have never integrated GA for organice. Like I said, I only found out about its integration very recently.

**Comment by alphapapa** [comment]
Hi Alain,

(I feel like I know you a little bit from watching that video presentation you put on YouTube where you demonstrated many things about Emacs.)

I appreciate your thorough response.  I'm not in a position to contribute to this project now, as I haven't actually used it yet, and I'm not familiar with the JavaScript frameworks and such that it uses.  And I'm busy with several Org-related projects of my own.  :)

I do look forward to watching your progress and trying it in the future.  For myself, having the analytics removed by default would probably be a prerequisite (I don't mean that as a demand, just my two cents).

Thanks.

**Comment by munen** [comment]
Dear @alphapapa 

First, let me say thank you for watching the presentation! I entered
that talk, as well as an intro to organice, into the call for
proposals for EmacsConf 2019. Today, my business partner (@branch14)
and I decided we're going to host a satellite in Zurich, Switzerland. If you're close (at all), you're happily invited, of course!

> I appreciate your thorough response

I do appreciate your clear words on this topic. I must have not been very awake when I saw that GA is integrated. As much as I usually ramble about Google, I really had a lapse of consciousness here. 

>  I'm not in a position to contribute to this project now

You did. Big time - you woke me up, thank you! :pray:

I'm happy to announce that as of now, organice is free of Google Analytics: 2e7b31c0c0eef5f3e3cac623c0a4af6fd9546019

> I do look forward to watching your progress and trying it in the future.

Please continue filing issues for whatever seems off or needs improving. Your time and effort is greatly appreciated!

> And I'm busy with several Org-related projects of my own. :)

I did look into them and am happily impressed. Lots of elisp goodness to integrate into [my Emacs configuration](https://github.com/munen/emacs.d/). There's loads that looks amazing - I'll probably start with integrating org-super-agenda. 

Seeing that you're heavily invested in writing good tooling on top of Org, maybe you're interested in one of the side-projects we just started. We're writing a proper BNF based parser for Org. Our initial use-case is to integrate it into organice, but the parser can be integrated into any JavaScript or Java codebase, later. If you're interested, we just got started with it: https://gitlab.200ok.ch/200ok/org-parser

Again, thank you very much for the issue and your other open source contributions! \(^_^)/ 

If you're satisfied with my fix, please feel free to close the issue.

**Comment by munen** [comment]
Dear @alphapapa 

I'm assuming you missed the notification (or didn't have the time) [9 days ago](https://github.com/200ok-ch/organice/issues/41#issuecomment-529028703). I'm closing this issue in the good hopes that it has been resolved to your liking.

If I'm mistaken, let's re-open and have a discussion^^

Thank you very much for your contribution to organice!

Best regards
Alain
